From c5acdc6b5da9e73b83ba3a2fd1dbd3fea9b66c0a Mon Sep 17 00:00:00 2001
From: Bing Li <sarah.libing@gmail.com>
Date: Tue, 7 Jul 2015 23:56:00 -0800
Subject: [PATCH 0518/1164] CDH-34269: HIVE-11201 : HCatalog  is ignoring user
 specified avro schema in the table definition
 (Bing Li via Ashutosh Chauhan)

Change-Id: Ic82cbb495fe1c16de53fd1d678831589a765fd8c
Signed-off-by: Ashutosh Chauhan <hashutosh@apache.org>
---
 .../hive/hcatalog/mapreduce/SpecialCases.java      |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/hcatalog/core/src/main/java/org/apache/hive/hcatalog/mapreduce/SpecialCases.java b/hcatalog/core/src/main/java/org/apache/hive/hcatalog/mapreduce/SpecialCases.java
index f38d53b..c536dd4 100644
--- a/hcatalog/core/src/main/java/org/apache/hive/hcatalog/mapreduce/SpecialCases.java
+++ b/hcatalog/core/src/main/java/org/apache/hive/hcatalog/mapreduce/SpecialCases.java
@@ -112,14 +112,14 @@ public static void addSpecialCasesParametersToOutputJobProperties(
         colTypes.add(TypeInfoUtils.getTypeInfoFromTypeString(field.getTypeString()));
       }
 
-      jobProperties.put(AvroSerdeUtils.AvroTableProperties.SCHEMA_LITERAL.getPropName(),
+      if (jobProperties.get(AvroSerdeUtils.AvroTableProperties.SCHEMA_LITERAL.getPropName())==null
+          || jobProperties.get(AvroSerdeUtils.AvroTableProperties.SCHEMA_LITERAL.getPropName()).isEmpty()) {
+     
+        jobProperties.put(AvroSerdeUtils.AvroTableProperties.SCHEMA_LITERAL.getPropName(),
           AvroSerDe.getSchemaFromCols(properties, colNames, colTypes, null).toString());
-
-
-      for (String propName : jobProperties.keySet()){
-        String propVal = jobProperties.get(propName);
       }
 
+
     }
   }
 
-- 
1.7.9.5

